Design/Instrument-2.0/Detector_Scan_NeXus_Saving.md

```diff
@@ -0,0 +1,129 @@
+
+#### Differences Between NeXus and Mantid Approach
+
+In the `DetectorInfo` representation in Mantid each position, rotation and mask flag of each pixel at each time index is stored separately. The scans are done on a pixel-by-pixel basis, and the components can not currently do any scanning. Different pixels are allowed to be given different scan intervals even within a component.
+
+The NeXus instrument proposal would not support scanning at the pixel level. The offset entries would be the same for each time index, with the position entries giving the scan for a component.
+
+In the case of writing raw NeXus files from the instrument it is likely you would only need a subset of the offset, position scan and rotation entries based on physical knowledge of the instrument. After processing with Mantid this might not be so easy to achieve, as detectors can be moved around in an arbitrary way, e.g. for calibration.
+
+The NeXus proposal for scanning allows the start and end positions and rotations to be different during a scan interval. This is not supported in Mantid.
+
+### Component Scanning
+
+_Based on discussion with Simon Heybrock._
+
+To resolve the differences between the NeXus proposal and Mantid one option is to allow components to scan. This is similar to how `RectangularDetector` works. The changes required would be:
+
+* Add the scan interface in `DetectorInfo` to `ComponentInfo`
+  * E.g. in addition to `ComponentInfo::setPosition(detIndex)` we have `ComponentInfo::setPosition({detIndex, timeIndex})`
+  * A `ComponentInfo::setPosition()` call updates all of the child positions (same for rotations)
+* For setting positions, rotations or mask that can change with the time index, both `ComponentInfo` and `DetectorInfo` will require a check to see if the parent is scanning
+  * If the parent is scanning it should throw
+* The in memory representation is the same as before - the absolute position and rotation of all pixels are stored
+
+The performance impact of this is that every call to `DetectorInfo::position()` will have an extra branch, even for non-scanning workspaces.
+
+An alternative approach is to have something similar to `RectangularDetectors`. Here instead of storing the scan positions for each detector we would just store the position for the parent. Positions would need to computed on the fly to get the actual position for each detector. This would have the advantage of reducing memory use*, at the cost of more CPU work. This would be a step away from some of the Instrument 2.0 design principles.
+
+*Note that the memory overhead due to scanning is not the main issue for the ILL instruments, addressing the overhead from using histograms with a  single count (no ToF) is a higher priority.
+
+#### Outstanding Questions
+
+* If components scan, are detectors still allowed to scan independently? 
```

Comment:
Some instrument definitions, see Polref for example, although true also of anything with monitors, has detectors that are not attached to any higher level components. That should be considered.

Reply:
Yes, that should definitely be allowed, the monitor case is a clear example.